src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -234,13 +237,26 @@ private function appendToken(SessionInterface $sessionContainer, Response $respo
             return FigResponseCookies::set($response, $this->getExpirationCookie());
         }
 
-        if (! $sessionContainerChanged) {
-            return $response;
+        if ($sessionContainerChanged || $this->shouldTokenBeRefreshed($token)) {
+            return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        }
+
+        return $response;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    private function shouldTokenBeRefreshed(Token $token = null) : bool
+    {
+        if (null === $token) {
+            return false;
         }
 
-        return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        return time() >= ($token->getClaim('iat') + $this->refreshTime);
```

Comment:
Now our problem lies on the ```testWillIgnoreUnSignedTokens``` case.

We can change the claim values but that's just a distraction.
What is really happening is that we're considering an unsigned token - or a token with invalid signature - a valid thing to be used, and with that we're appending the token cookie (and returning a new response instance).

A nasty thing that can be done is to use references to invalidate the token instance when the signature is invalid, but if the callback that creates the ```SessionInterface``` is not called we'll probably append the cookie.

Any thoughts?